pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -398,6 +400,55 @@ private void ensureMinReplicas(TableConfig config) {
     }
   }
 
+  /**
+   * Verify table configs if it's a hybrid table, i.e. having both offline and real-time sub-tables.
+   */
+  private void verifyTableConfigs(TableConfig newTableConfig) {
+    String rawTableName = TableNameBuilder.extractRawTableName(newTableConfig.getTableName());
+    LOGGER.info("Validating table configs for Table: {}", rawTableName);
+
+    TableConfig tableConfigToCompare = null;
+    if (newTableConfig.getTableType() == CommonConstants.Helix.TableType.REALTIME) {
+      if (_pinotHelixResourceManager.hasOfflineTable(rawTableName)) {
+        tableConfigToCompare = _pinotHelixResourceManager.getOfflineTableConfig(rawTableName);
+      }
+    } else {
+      if (_pinotHelixResourceManager.hasRealtimeTable(rawTableName)) {
+        tableConfigToCompare = _pinotHelixResourceManager.getRealtimeTableConfig(rawTableName);
+      }
+    }
+
+    // Check if it is a hybrid table or not. If not, there's no need to verify anything.
+    if (tableConfigToCompare == null) {
+      LOGGER.info(
+          "Table: {} is not a hybrid table. Skipping consistency check across realtime and offline parts of the table.",
+          rawTableName);
+      return;
+    }
+
+    SegmentsValidationAndRetentionConfig newSegmentConfig = newTableConfig.getValidationConfig();
+    SegmentsValidationAndRetentionConfig SegmentConfigToCompare = tableConfigToCompare.getValidationConfig();
+
+    String newTimeColumnName = newSegmentConfig.getTimeColumnName();
+    String existingTimeColumnName = SegmentConfigToCompare.getTimeColumnName();
+    if (!existingTimeColumnName.equals(newTimeColumnName)) {
+      throw new PinotHelixResourceManager.InvalidTableConfigException(
+          String.format("Time column names are different! Existing time column name: %s. New time column name: %s",
+              existingTimeColumnName, newTimeColumnName));
+    }
+
+    String newTimeColumnType = newSegmentConfig.getTimeType();
+    String existingTimeColumnType = SegmentConfigToCompare.getTimeType();
+    if (!existingTimeColumnType.equals(newTimeColumnType)) {
```

Comment:
this can be equalsIgnoreCase, unless you are using to get the TimeUtils and turn it into an enum before comparison.